### PR TITLE
e2e: stoping using deprecated PollImmediateWithContext

### PR DESCRIPTION
**Pull request by wlq1212**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:

e2e: stoping using deprecated PollImmediateWithContext

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```
NONE
```

### CODE REVIEW

File: test/e2e/apimachinery/crd_conversion_webhook.go [hunk 1]
```diff
@@ -478,7 +479,7 @@ func testCRListConversion(ctx context.Context, f *framework.Framework, testCrd *
 
 // waitWebhookConversionReady sends stub custom resource creation requests requiring conversion until one succeeds.
 func waitWebhookConversionReady(ctx context.Context, f *framework.Framework, crd *apiextensionsv1.CustomResourceDefinition, customResourceClients map[string]dynamic.ResourceInterface, version string) {
-	framework.ExpectNoError(wait.PollImmediateWithContext(ctx, 100*time.Millisecond, 30*time.Second, func(ctx context.Context) (bool, error) {
+	framework.ExpectNoError(wait.PollUntilContextTimeout(ctx, 100*time.Millisecond, 30*time.Second, false, func(ctx context.Context) (bool, error) {
```
Comment (kerthcet) [inline test/e2e/apimachinery/crd_conversion_webhook.go:482]:
```suggestion
	framework.ExpectNoError(wait.PollUntilContextTimeout(ctx, 100*time.Millisecond, 30*time.Second, true, func(ctx context.Context) (bool, error) {
```

It should be true here because we want to pull **immediately**. Apply to other files also. Thanks.

Comment (wlq1212) [reply test/e2e/apimachinery/crd_conversion_webhook.go]: done

**Comment by pohly** [comment]
@wlq1212 : did you notice the pull-kubernetes-linter-hints failure?

The missing error checking reported there is a genuine problem that also occurred in the old code. Because you are touching it, it gets reported.

**Comment by jiahuif** [comment]
/assign @kerthcet
Thank you.
/triage accepted

**Comment by wlq1212** [comment]
false | /test pull-kubernetes-e2e-gci-gce-ipvs
-- | --

**Comment by pohly** [comment]
I'm undecided whether replacing `wait.PollImmediateWithContext` with `wait.PollUntilContextTimeout` is really making the tests sufficiently better to justify the code churn. What really would make a difference would be to remove `wait` altogether (see https://github.com/kubernetes/kubernetes/issues/106575), but that is significantly more work.

I'd like to hear from others before approving this PR.

**Comment by pohly** [comment]
https://pkg.go.dev/k8s.io/apimachinery/pkg/util/wait#PollImmediateWithContext says:

> Will be removed in a future release. 

That means we should be removing its usage.

File: test/e2e/apimachinery/crd_conversion_webhook.go [hunk 2]
```diff
@@ -34,12 +34,13 @@ import (
...
 	clientset "k8s.io/client-go/kubernetes"
+	admissionapi "k8s.io/pod-security-admission/api"
+	"k8s.io/utils/pointer"
+
 	"k8s.io/kubernetes/test/e2e/framework"
 	e2edeployment "k8s.io/kubernetes/test/e2e/framework/deployment"
 	"k8s.io/kubernetes/test/utils/crd"
 	imageutils "k8s.io/kubernetes/test/utils/image"
-	admissionapi "k8s.io/pod-security-admission/api"
-	"k8s.io/utils/pointer"
```
Comment (bart0sh) [inline test/e2e/apimachinery/crd_conversion_webhook.go:42]: How moving these imports around is related to the PR?

**Comment by bart0sh** [comment]
@pohly 
>  What really would make a difference would be to remove `wait` altogether (see #106575), but that is significantly more work.

As far as I understand, #106675 is about testing code. I agree that your suggestion to use `gomega.Eventually` makes a lot of sense there. How would you suggest to get rid of `wait` in non-testing  code?

@wlq1212 Please, fix CI test failures and address remaining comments, thanks.

/cc

**Comment by pohly** [comment]
> How would you suggest to get rid of wait in non-testing code?

It's fine to use the `wait` package in non-test code.

**Comment by kerthcet** [comment]
> I'm undecided whether replacing `wait.PollImmediateWithContext` with `wait.PollUntilContextTimeout` is really making the tests sufficiently better to justify the code churn. What really would make a difference would be to remove `wait` altogether (see #106575), but that is significantly more work.
> 
> I'd like to hear from others before approving this PR.

Yes, that's a significantly more work, but I think there's no reason to block this since it counts for the code cleanliness. By the way, do we have any plan to remove the `wait` for sig-testing, avoiding the repetitive work round by round.

**Comment by pohly** [comment]
>  think there's no reason to block this since it counts for the code cleanliness.

I wasn't suggesting to block this PR. I just wanted to clarify whether it's worth merging and [later found that it is](https://github.com/kubernetes/kubernetes/pull/120303#issuecomment-1708893219).

> do we have any plan to remove the wait for sig-testing, avoiding the repetitive work round by round.

No specific plan, just an open issue.

File: staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/controller/controller.go [hunk 3]
```diff
@@ -23,13 +23,14 @@ import (
 	"time"
 
 	"github.com/fsnotify/fsnotify"
+	"k8s.io/klog/v2"
```
Comment (aramase) [inline staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/controller/controller.go:26]: Please undo this change.
